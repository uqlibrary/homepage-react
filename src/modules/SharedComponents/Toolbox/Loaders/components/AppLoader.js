import React from 'react';
import PropTypes from 'prop-types';
import CircularProgress from '@mui/material/CircularProgress';
import Typography from '@mui/material/Typography';
import Grid from '@mui/material/Grid';
import { styled } from '@mui/material/styles';

const StyledLoader = styled(Grid)(({ theme }) => ({
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    ...(((theme.palette || {}).primary || {}).gradient || {}).diagonal,
    width: '100%',
    height: '100%',
    textAlign: 'center !important',

    '& .white': {
        color: ((theme.palette || {}).white || {}).main,
        fontWeight: (theme.typography || {}).fontWeightLight,
    },
    '& .spaceBetween': {
        margin: '16px 0',
    },
    '& .logo': {
        width: 200,
    },
}));

export class AppLoader extends React.Component {
    static propTypes = {
        title: PropTypes.string.isRequired,
        logoImage: PropTypes.string,
        logoText: PropTypes.string,
    };

    render() {
        const { title, logoImage, logoText } = this.props;
        return (
            <StyledLoader container spacing={0} direction="column" justifyContent="center" alignItems="center">
                <Grid item className={'spaceBetween'}>
                    <CircularProgress
                        size={80}
                        thickness={1}
                        className={'white'}
                        aria-label="Loading Library website"
                    />
                </Grid>
                <Grid item className={'spaceBetween'}>
                    {logoImage && <div className={`${logoImage} logo`} alt={logoText} />}
                </Grid>
                <Grid item className={'spaceBetween'}>
                    <Typography variant={'h6'} className={'white'}>
                        {title}
                    </Typography>
                </Grid>
            </StyledLoader>
        );
    }
}

export default AppLoader;
