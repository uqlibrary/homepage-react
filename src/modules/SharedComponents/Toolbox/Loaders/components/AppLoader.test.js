import { styles } from './AppLoader';

describe('AppLoader component', () => {
    it('should have a proper style generator', () => {
        const theme = {
            palette: {
                primary: {
                    gradient: {
                        diagonal: {
                            test1: 'test1',
                            test2: 'test2',
                            test3: 'test3',
                        },
                    },
                },
                white: {
                    main: 'test4',
                },
            },
            typography: {
                fontWeightLight: 'test5',
            },
        };
        expect(styles(theme)).toMatchSnapshot();

        delete theme.palette.primary;
        expect(styles(theme)).toMatchSnapshot();

        delete theme.palette;
        expect(styles(theme)).toMatchSnapshot();

        delete theme.typography;
        expect(styles(theme)).toMatchSnapshot();
    });
});
