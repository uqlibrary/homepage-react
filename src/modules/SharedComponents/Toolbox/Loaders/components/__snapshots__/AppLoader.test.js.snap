// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLoader component should have a proper style generator 1`] = `
Object {
  "appLoader": Object {
    "alignItems": "center",
    "display": "flex",
    "height": "100%",
    "justifyContent": "center",
    "test1": "test1",
    "test2": "test2",
    "test3": "test3",
    "textAlign": "center !important",
    "width": "100%",
  },
  "logo": Object {
    "width": 200,
  },
  "spaceBetween": Object {
    "margin": "16px 0",
  },
  "white": Object {
    "color": "test4",
    "fontWeight": "test5",
  },
}
`;

exports[`AppLoader component should have a proper style generator 2`] = `
Object {
  "appLoader": Object {
    "alignItems": "center",
    "display": "flex",
    "height": "100%",
    "justifyContent": "center",
    "textAlign": "center !important",
    "width": "100%",
  },
  "logo": Object {
    "width": 200,
  },
  "spaceBetween": Object {
    "margin": "16px 0",
  },
  "white": Object {
    "color": "test4",
    "fontWeight": "test5",
  },
}
`;

exports[`AppLoader component should have a proper style generator 3`] = `
Object {
  "appLoader": Object {
    "alignItems": "center",
    "display": "flex",
    "height": "100%",
    "justifyContent": "center",
    "textAlign": "center !important",
    "width": "100%",
  },
  "logo": Object {
    "width": 200,
  },
  "spaceBetween": Object {
    "margin": "16px 0",
  },
  "white": Object {
    "color": undefined,
    "fontWeight": "test5",
  },
}
`;

exports[`AppLoader component should have a proper style generator 4`] = `
Object {
  "appLoader": Object {
    "alignItems": "center",
    "display": "flex",
    "height": "100%",
    "justifyContent": "center",
    "textAlign": "center !important",
    "width": "100%",
  },
  "logo": Object {
    "width": 200,
  },
  "spaceBetween": Object {
    "margin": "16px 0",
  },
  "white": Object {
    "color": undefined,
    "fontWeight": undefined,
  },
}
`;
